### PR TITLE
Pino 7 new transport interface

### PR DESCRIPTION
**Pull request by Eomm**
Ref https://github.com/pinojs/pino/issues/1056

Main changes:

- removed `useUnifiedTopology` and `useNewUrlParser` options. Since [mongo client v4 ](https://github.com/mongodb/node-mongodb-native/blob/0dc9b4531d8ca4a4b37a0a4d53b992219b802fb7/docs/CHANGES_4.0.0.md#mongoclientoptions-interface) these options are ignored - this module uses the 4.1 sticky version
- expose the transport interface using the `pino-abstract-transport` utility
- implement the `cli`'s logic as new pino v7 transport

Do you have some tests to suggest?
Do you think that we could use the mongo's bulk write to insert rows? - if so, what frequency?

Todo:

- [x] documentation

### CODE REVIEW

File: test/transport.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  })
+  const log = pino(transport)
+  await once(transport, 'ready')
+  log.info('this is a string log')
+  log.debug('ignored')
+  log.info('this is a long string log'.repeat(1000))
+  log.fatal(new Error('ops'), 'not ignored')
+  t.pass('logged on mongo')
+
+  await setTimeout(1000)
```
Comment (Eomm) [inline test/transport.js:45]: Here, I did not found a way to wait `pino.flush()` (and it was triggering an error. I'm investigating)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -17,6 +17,15 @@ jobs:
         node-version: [12, 14, 16]
         os: [macos-latest, ubuntu-latest, windows-latest]
 
+    services:
+      mongodb:
+        image: mongo:5.0.0
```
Comment (kibertoad) [inline .github/workflows/ci.yml:22]: Why not 5.0.2?

Comment (Eomm) [reply .github/workflows/ci.yml]: copy-paste

Fixed

File: test/end-to-end/transport.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  await once(transport, 'ready')
+  log.info('this is a string log')
+  log.debug('ignored')
+  log.info('this is a long string log'.repeat(1000))
+  log.fatal(new Error('ops'), 'not ignored')
+  t.pass('logged on mongo')
+
+  await new Promise(resolve => {
+    setTimeout(resolve, 1000)
+  })
```
Comment (mcollina) [inline test/end-to-end/transport.js:46]: please use promisify instead.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -15,10 +15,9 @@ jobs:
     strategy:
       matrix:
         node-version: [12, 14, 16]
-        os: [macos-latest, ubuntu-latest, windows-latest]
+        os: [macos-latest, windows-latest]
```
Comment (mcollina) [inline .github/workflows/ci.yml:18]: if you want to drop one, drop mac, not Linux.

Comment (kibertoad) [reply .github/workflows/ci.yml:18]: osn't the idea that linux is civered by e2e Docker ones?

Comment (Eomm) [reply .github/workflows/ci.yml:18]: From GH error

> Container operations are only supported on Linux runners

So there is a new job that run on Linux and execute the old test plus the new ones

File: lib/pino-transport.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+const { Writable } = require('stream')
+const { MongoClient } = require('mongodb')
+const parseMongoUrl = require('muri')
```
Comment (climba03003) [inline lib/pino-transport.js:5]: I think you need to remove `muri` as a dependency as it block the usage for `MongoDB Atlas` which have a protocol `mongodb+srv`.

Related Issue: #14

As `client.db()` will use the database name in the `uri`, we can use pass `database` option inside.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
there appear to be some conflicts

**Comment by Eomm** [comment]
> there appear to be some conflicts

Done

**Pull request merged**